### PR TITLE
Mystify scrolls and wands when GM drops spell onto an actor sheet while holding the ALT key

### PR DESCRIPTION
**Pull request by simonward**

### CODE REVIEW

**Comment by CarlosFdez** [comment]
Would it not work better to move all these checks to _onDropItemCreate?

**Comment by simonward** [comment]
> Would it not work better to move all these checks to _onDropItemCreate?

We'd need to override _onDropItemCreate(), and there's no event parameter so we'd need to examine the keyboard state.
If that sounds reasonable, I can take a look at implementing it that way.

**Comment by CarlosFdez** [comment]
This is a good point. My thought now is to assign `const mystify = game.user.isGM && event.altKey;` and either check that or pass to the createSpellFromConsumableFunction as a new parameter (convert to options parameter).

In fact, suspiciously, further up the mystify already there doesn't check if they're a GM, when it probably should.

File: src/module/actor/sheet/base.ts [hunk 1]
```diff
@@ -1061,6 +1061,12 @@ abstract class ActorSheetPF2e<TActor extends ActorPF2e> extends ActorSheet<TActo
                     {},
                     async (heightenedLevel, itemType, spell) => {
                         const createdItem = await createConsumableFromSpell(itemType, spell, heightenedLevel);
```
Comment (stwlam) [inline src/module/actor/sheet/base.ts:1063]: Do it from this function:

1. Convert the parameters to a destructured object.
2. Add a `mystify` boolean property.

**Review by CarlosFdez — DISMISSED** [review]
If I'd change anything, its that I'd make spell a non-options paramater, but for me that's not a hard or fast thing, this is perfectly fine.

**Pull request merged**